galleries/plot_types/basic/plot.py

```diff
@@ -13,12 +13,14 @@
 
 # make data
 x = np.linspace(0, 10, 100)
-y = 4 + 2 * np.sin(2 * x)
+y = 4 + 1 * np.sin(2 * x)
 
 # plot
 fig, ax = plt.subplots()
 
+ax.plot(x[::4], y[::4]+2.5, 'x', markeredgewidth=2)
 ax.plot(x, y, linewidth=2.0)
+ax.plot(x[::4], y[::4]-2.5, 'o-', linewidth=2)
```

Comment:
I'm not a fan of markevery. You typically should not throw away your data.

In particular, a strong case for markers-only is that you don't want to imply continuity (I have on data in between, so shouldn't connect data points; nobody knows what's happening there). Then using *markevery*, which is only meaningful under the assumption of continuity, is inconsistent.

Either way, `[::4]` and `markevery` are only hacks to get less data points. I see a small advantage for `[.::4]` because it reduces the data before `plot` and we don't imply the markevery semantics. I've switched to using separate data `x2, y2` now. While that costs us two extra lines to generate the data, it's less hacky because it does not reuse parts of the previous data.

Reply:
If you're plotting a solid line *and* markers (the bottom case) wouldn't it be better to use `markerevery` there? Because there's a line it implies continuity, but you only want to put markers at certain points on that line.

Reply:
I have strong reservations against markevery. Generally, why would you randomly draw only some of your data points? Usually, you’d either draw all markers, or if that’s too dense, switch to line-only. IMHO markevery is only for *very* specific situations.

In contrast, with the fewer points given, it can still make sense to draw a line on top of markers if you want to imply continuity and guide the eye.

Reply:
Different shaped markers can be added to solid lines to distinguish them, if you have too many lines to only distinguish by colour.  If the actual data points are dense, markerevery would make sense in that case.

Reply:
Yes, I count this under “very specific situations “. 😁